tenders/hvt/hvt_module_blk.c

```diff
@@ -70,7 +69,11 @@ static void hypercall_block_write(struct hvt *hvt, hvt_gpa_t gpa)
 
     ret = pwrite(e->b.hostfd, HVT_CHECKED_GPA_P(hvt, wr->data, wr->len),
             wr->len, pos);
-    assert(ret == wr->len);
+    if (ret != wr->len) {
+        /* Report that it failed somehow */
+        wr->ret = SOLO5_R_EUNSPEC;
+        return;
+    }
```

Comment:
This changes the semantics so the unikernel can continue if the write fails. Which is perhaps okay. It's hard for the unikernel to act on it, though(?!)

From what I can read from `man -s2 pwrite write` it can return less if there is not enough disk space. Considering the file is checked to have a fixed size from the beginning I can only imagine this happening if the file is a) sparse, or b) has been truncated(!)

Reply:
It is not entirely clear to me whether it is worth to handle this situation gracefully. What is wrong about assertion failure and the unikernel exiting?

Handling this properly from the unikernel side may require some further information, i.e. transport the data from errno from the tender to the unikernel, and then deciding what to do in the unikernel.

Reply:
Thanks, I only put a minor comment to use `strerror()`